ui-tests/cypress/integration/pages/templates.spec.js

```diff
@@ -1,142 +1,621 @@
+          page.getContent().getTemplateValue().then(value => expect(value).to.equal(sampleData.template));
+          page.getContent().getPreviewGrid().should('exist').and('be.visible');
+        });
     });
 
-    it('If the "descr" property of any object in the JSON configuration is an object, it should accept. Other mis-types under "descr" will display an error beneath (ENG-2711)', () => {
-      cy.pageTemplatesController()
-        .then(controller => controller.addPageTemplate({
-          ...sampleData,
-          configuration: JSON.parse(sampleData.configuration)
-        }));
-      cy.wrap(sampleData.code).as('templateToBeDeleted');
-      currentPage = openPageTemplateMgmtPage();
-      currentPage = currentPage.getContent().getKebabMenuByCode(sampleData.code).open().openEdit();
-      currentPage.getContent().getJsonConfigInput().click();
-      cy.realPress(['Meta', 'F']);
-      cy.realType('"descr":{enter}{rightarrow}{rightarrow}');
-      cy.realPress(['Meta', 'Shift', 'ArrowRight']);
-      cy.realType('{backspace}');
-      cy.realType('{', {parseSpecialCharSequences: false});
-      currentPage.getContent().getJsonConfigArea().invoke('hasClass', 'has-error').should('be.true');
-      cy.realType('},', {parseSpecialCharSequences: false});
-      currentPage.getContent().getJsonConfigArea().invoke('hasClass', 'has-error').should('be.false');
+  });
+
+  describe('Templates list', () => {
+
+    it([Tag.SANITY, 'ENG-3525'], 'Default page templates', () => {
+      openPageTemplateMgmtPage()
+        .then(page => {
+          cy.validateUrlPathname('/page-template');
+          page.getContent().getTable().should('exist').and('be.visible')
+              .then(table => {
+                cy.wrap(table.children(htmlElements.tbody).find(htmlElements.tr))
+                  .then(rows => cy.validateListTexts(rows, defaultTemplates.map(template => template.code+template.descr+'EditCloneDetailsDelete')));
+              });
+            });
     });
+
+    it([Tag.SANITY, 'ENG-3525'], 'The templates list pagination should be correct', () => {
+      addRandomPageTemplates(4);
+
+      openPageTemplateMgmtPage()
+        .then(page => {
+          cy.validateUrlPathname('/page-template');
+          page.getContent().getTable().should('exist').and('be.visible')
+              .then(table => {
+                cy.wrap(table.children(htmlElements.tbody).find(htmlElements.tr))
+                  .then(rows => {
+                    page.getContent().getPagination().getDropdownButton().should('have.text', rows.length+' ');
+                  });
+              });
+          page.getContent().getPagination().getInput().should('have.value', 1);
+        });
+    });
+
+    describe('Navigating the pagination', () => {
+
+      before(() => {
+        cy.kcAPILogin();
+        cy.wrap([]).as('templatesToBeDeleted');
+        addRandomPageTemplates(14);
+      });
+
```

Comment:
I used a different way to delete the created templates here, so that the added templates could be created just once instead of having a number of POST and DELETE requests before and after each of the tests that need to have a filled up list

Reply:
Aliases are reset before each test, so it wouldn't be possible to retrieve the saved codes after the first test

Reply:
the `afterEach` should delete all the templates added to the _alias_ at the end of the first test, how is this gonna work?
If you store this values in a separate _alias_, you should be able to access them via `this` properties even if the alias is reset at the end of each test